core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/EraSubmissionPreparation.java

```diff
@@ -0,0 +1,496 @@
+      sampleName.appendChild(sampleScientificName);
+
+      // 2/11/2011 Antony Colles moved IF !=null statement, to help produce valid submission XML.
+      Element sampleTaxonIdentifier = xml.getOwnerDocument().createElementNS(null, "TAXON_ID");
+      if (!isStringEmptyOrNull(sample.getTaxonIdentifier())) {
+        sampleTaxonIdentifier.setTextContent(sample.getTaxonIdentifier());
+      } else {
+        sampleTaxonIdentifier.setTextContent("000001");
+      }
+      sampleName.appendChild(sampleTaxonIdentifier);
+
+      xml.appendChild(sampleName);
+
+      Element sampleDescription = xml.getOwnerDocument().createElementNS(null, "DESCRIPTION");
+      sampleDescription.setTextContent(sample.getDescription());
+      xml.appendChild(sampleDescription);
+
+    }
+
+    @Override
+    protected String xmlCollectionName() {
+      return "SAMPLE_SET";
+    }
+
+    @Override
+    protected String xmlName() {
+      return "SAMPLE";
+    }
+  }
+
+  private final class StudyXmlSubfile extends XmlSubmissionFromSet<Study> {
+    @Override
+    public Stream<Study> items() {
+      return submission.getExperiments().stream().map(Experiment::getStudy).distinct();
+    }
+
+    @Override
+    public String name() {
+      return "study";
+    }
+
+    @Override
+    protected void populate(Element xml, Study s) {
+      xml.setAttribute("alias", s.getAlias());
+
+      Element studyDescriptor = xml.getOwnerDocument().createElementNS(null, "DESCRIPTOR");
+      xml.appendChild(studyDescriptor);
+
+      Element studyTitle = xml.getOwnerDocument().createElementNS(null, "STUDY_TITLE");
+      studyTitle.setTextContent(s.getAlias());
+      studyDescriptor.appendChild(studyTitle);
+
+      Element studyType = xml.getOwnerDocument().createElementNS(null, "STUDY_TYPE");
+      studyType.setAttribute("existing_study_type", s.getStudyType().getName());
+      studyDescriptor.appendChild(studyType);
+
+      Element centerProjectName = xml.getOwnerDocument().createElementNS(null, "CENTER_PROJECT_NAME");
+      centerProjectName.setTextContent(s.getProject().getAlias());
+      studyDescriptor.appendChild(centerProjectName);
+
+      Element studyAbstract = xml.getOwnerDocument().createElementNS(null, "STUDY_ABSTRACT");
+      studyDescriptor.appendChild(studyAbstract);
+
+      Element studyDescription = xml.getOwnerDocument().createElementNS(null, "STUDY_DESCRIPTION");
+      studyDescription.setTextContent(s.getDescription());
+      studyDescriptor.appendChild(studyDescription);
+    }
+
+    @Override
+    protected String xmlCollectionName() {
+      return "STUDY_SET";
+    }
+
+    @Override
+    protected String xmlName() {
+      return "STUDY";
+    }
+  }
+
+  private abstract class XmlSubmissionFromSet<T> extends ChildSubmissionFile {
+    @Override
+    public void generateDocument(ZipOutputStream output)
+        throws ParserConfigurationException, IOException, TransformerConfigurationException, TransformerException,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'javax.xml.transform.TransformerConfigurationException' which is a subclass of 'javax.xml.transform.TransformerException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)